<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0" colspan="2">
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QPushButton" name="addThreadProfile">
        <property name="text">
         <string>Добавить профиль
 разделки</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="comboBox_2">
        <item>
         <property name="text">
          <string>36.txt</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>37.txt</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>58.txt</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>90.txt</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>91.txt</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>92.txt</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="addRandomGraph">
        <property name="text">
         <string>Добавить
рандомный график</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="selectRegion">
        <property name="text">
         <string>Выделить
область на графике</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="removeSelectionRegion">
        <property name="text">
         <string>Убрать область</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="zoomProfile">
        <property name="text">
         <string>Zoom 1:1</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item row="1" column="0">
     <widget class="QCustomPlot" name="customPlot" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QPushButton" name="pushButton">
        <property name="text">
         <string>Аппроксимация
выделенных точек
прямой</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pushButton_2">
        <property name="text">
         <string>Убрать линии</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QLabel" name="label">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Количество
точек для
аппроксимации</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="spinBox">
          <property name="minimum">
           <number>5</number>
          </property>
          <property name="value">
           <number>20</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Угол перегиба</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="doubleSpinBox">
          <property name="decimals">
           <number>1</number>
          </property>
          <property name="minimum">
           <double>1.000000000000000</double>
          </property>
          <property name="maximum">
           <double>45.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
          <property name="value">
           <double>3.000000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QLabel" name="label_7">
        <property name="frameShape">
         <enum>QFrame::Box</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Plain</enum>
        </property>
        <property name="text">
         <string>Результаты расчета:</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <item>
         <widget class="QLabel" name="label_4">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>50</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>Угол
разделки</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTextBrowser" name="grooveAngle">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>50</width>
            <height>20</height>
           </size>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_7">
        <item>
         <widget class="QLabel" name="label_5">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>50</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>Смещение
кромок</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTextBrowser" name="edgeDisplacement">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>50</width>
            <height>20</height>
           </size>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_8">
        <item>
         <widget class="QLabel" name="label_6">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>50</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>Зазор</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTextBrowser" name="gapValue">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>50</width>
            <height>20</height>
           </size>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QCheckBox" name="checkBox">
        <property name="acceptDrops">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string comment="При отключении фильтра группы точек сбрасываются" extracomment="При отключении фильтра группы точек сбрасываются"/>
        </property>
        <property name="toolTipDuration">
         <number>5</number>
        </property>
        <property name="statusTip">
         <string comment="При отключении фильтра группы точек сбрасываются" extracomment="При отключении фильтра группы точек сбрасываются"/>
        </property>
        <property name="whatsThis">
         <string comment="При отключении фильтра группы точек сбрасываются" extracomment="При отключении фильтра группы точек сбрасываются"/>
        </property>
        <property name="layoutDirection">
         <enum>Qt::LeftToRight</enum>
        </property>
        <property name="text">
         <string>Фильтр точек</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_9">
        <item>
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>Кол. точек
в группе (min)</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="minNumOfPointsInGroup">
          <property name="minimum">
           <number>3</number>
          </property>
          <property name="maximum">
           <number>15</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Тип разделки</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="comboBox">
          <property name="maximumSize">
           <size>
            <width>60</width>
            <height>16777215</height>
           </size>
          </property>
          <item>
           <property name="text">
            <string>V</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>J</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>K</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>II</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QPushButton" name="pushButton_3">
        <property name="text">
         <string>Определить разделку</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3"/>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>Меню</string>
    </property>
    <addaction name="action"/>
   </widget>
   <widget class="QMenu" name="menu_2">
    <property name="title">
     <string>Обработка данных</string>
    </property>
    <addaction name="selectKeyPoints"/>
    <addaction name="removeKeyPoints"/>
    <addaction name="buildApproxLines"/>
    <addaction name="removePointsLines"/>
    <addaction name="calculateParams"/>
    <addaction name="removeLines"/>
   </widget>
   <addaction name="menu"/>
   <addaction name="menu_2"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="action">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Добавить линию</string>
   </property>
  </action>
  <action name="action_2">
   <property name="text">
    <string>Построить отрезки между ключевыми точками</string>
   </property>
  </action>
  <action name="remove_keyPoints">
   <property name="text">
    <string>убрать ключевые точки</string>
   </property>
  </action>
  <action name="selectKeyPoints">
   <property name="text">
    <string>Выделить ключевые точки</string>
   </property>
  </action>
  <action name="removeKeyPoints">
   <property name="text">
    <string>Убрать ключевые точки</string>
   </property>
  </action>
  <action name="buildApproxLines">
   <property name="text">
    <string>Построить аппроксимированные прямые</string>
   </property>
  </action>
  <action name="removePointsLines">
   <property name="text">
    <string>Убрать точки и линии</string>
   </property>
  </action>
  <action name="calculateParams">
   <property name="text">
    <string>Расчёт параметров</string>
   </property>
  </action>
  <action name="removeLines">
   <property name="text">
    <string>Убрать линии</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
